model/event/subscriptions.go

```diff
@@ -0,0 +1,192 @@
+package event
+
+import (
+	"regexp"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	SubscriptionsCollection = "subscriptions"
+)
+
+var (
+	subscriptionIDKey             = bsonutil.MustHaveTag(Subscription{}, "ID")
+	subscriptionTypeKey           = bsonutil.MustHaveTag(Subscription{}, "Type")
+	subscriptionTriggerKey        = bsonutil.MustHaveTag(Subscription{}, "Trigger")
+	subscriptionSelectorsKey      = bsonutil.MustHaveTag(Subscription{}, "Selectors")
+	subscriptionRegexSelectorsKey = bsonutil.MustHaveTag(Subscription{}, "RegexSelectors")
+	subscriptionSubscriberKey     = bsonutil.MustHaveTag(Subscription{}, "Subscriber")
+
+	groupedSubscriberTypeKey       = bsonutil.MustHaveTag(GroupedSubscribers{}, "Type")
+	groupedSubscriberSubscriberKey = bsonutil.MustHaveTag(GroupedSubscribers{}, "Subscribers")
+
+	subscriberWithRegexKey               = bsonutil.MustHaveTag(SubscriberWithRegex{}, "Subscriber")
+	subscriberWithRegexRegexSelectorsKey = bsonutil.MustHaveTag(SubscriberWithRegex{}, "RegexSelectors")
+)
+
+type Subscription struct {
+	ID             bson.ObjectId `bson:"_id"`
+	Type           string        `bson:"type"`
+	Trigger        string        `bson:"trigger"`
+	Selectors      []Selector    `bson:"selectors,omitempty"`
+	RegexSelectors []Selector    `bson:"regex_selectors,omitempty"`
+	Subscriber     Subscriber    `bson:"subscriber"`
+}
+
+type Selector struct {
+	Type string `bson:"type"`
+	Data string `bson:"data"`
+}
+
+type GroupedSubscribers struct {
+	Type        string                `bson:"_id"`
+	Subscribers []SubscriberWithRegex `bson:"subscribers"`
+}
+
+type SubscriberWithRegex struct {
+	Subscriber     Subscriber `bson:"subscriber"`
+	RegexSelectors []Selector `bson:"regex_selectors"`
+}
+
+// FindSubscribers finds all subscriptions that match the given information
+func FindSubscribers(subscriptionType, triggerType string, selectors []Selector) ([]GroupedSubscribers, error) {
+	pipeline := []bson.M{
+		{
+			"$match": bson.M{
+				subscriptionTypeKey:    subscriptionType,
+				subscriptionTriggerKey: triggerType,
+			},
+		},
+		{
+			"$project": bson.M{
+				subscriptionSubscriberKey:     1,
+				subscriptionRegexSelectorsKey: 1,
+				"keep": bson.M{
+					"$and": []bson.M{
+						{
+							"$ne": []interface{}{
+								"$" + subscriptionSelectorsKey,
+								[]interface{}{},
+							},
+						},
+						{
+							"$setIsSubset": []interface{}{"$" + subscriptionSelectorsKey, selectors},
```

Comment:
should be a []bson.M

Reply:
this is a slice containing a string, and a slice of strings. This has to be []inteface{}.